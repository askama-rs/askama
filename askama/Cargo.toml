[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
