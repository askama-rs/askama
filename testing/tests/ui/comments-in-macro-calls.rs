use askama::Template;

// Regression test for <https://issues.oss-fuzz.com/issues/425145246>.

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(// hello) }}")]
struct LineComment;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(// hello)\n}}")]
struct LineComment2;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(/* hello) }}")]
struct BlockComment;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(/// hello) }}")]
struct OuterLineDoc;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(/// hello)\n}}")]
struct OuterLineDoc2;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(/** hello) }}")]
struct OuterBlockDoc;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(//! hello) }}")]
struct InnerLineDoc;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(//! hello)\n}}")]
struct InnerLineDoc2;

#[derive(Template)]
#[template(ext = "html", source = "{{ e!(/*! hello) }}")]
struct InnerBlockDoc;

fn main() {}
