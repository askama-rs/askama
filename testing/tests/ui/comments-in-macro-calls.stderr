error: you are probably missing a line break to end comment
 --> <source attribute>:1:6
       "// hello) }}"
 --> tests/ui/comments-in-macro-calls.rs:6:35
  |
6 | #[template(ext = "html", source = "{{ e!(// hello) }}")]
  |                                   ^^^^^^^^^^^^^^^^^^^^

error: expected `)` but found `}`
 --> <source attribute>:1:15
       "}}"
  --> tests/ui/comments-in-macro-calls.rs:10:35
   |
10 | #[template(ext = "html", source = "{{ e!(// hello)\n}}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^^

error: missing `*/` to close block comment
 --> <source attribute>:1:6
       "/* hello) }}"
  --> tests/ui/comments-in-macro-calls.rs:14:35
   |
14 | #[template(ext = "html", source = "{{ e!(/* hello) }}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^

error: you are probably missing a line break to end doc comment
 --> <source attribute>:1:6
       "/// hello) }}"
  --> tests/ui/comments-in-macro-calls.rs:18:35
   |
18 | #[template(ext = "html", source = "{{ e!(/// hello) }}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^^

error: expected `)` but found `}`
 --> <source attribute>:1:16
       "}}"
  --> tests/ui/comments-in-macro-calls.rs:22:35
   |
22 | #[template(ext = "html", source = "{{ e!(/// hello)\n}}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^^^

error: missing `*/` to close block doc comment
 --> <source attribute>:1:6
       "/** hello) }}"
  --> tests/ui/comments-in-macro-calls.rs:26:35
   |
26 | #[template(ext = "html", source = "{{ e!(/** hello) }}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^^

error: you are probably missing a line break to end doc comment
 --> <source attribute>:1:6
       "//! hello) }}"
  --> tests/ui/comments-in-macro-calls.rs:30:35
   |
30 | #[template(ext = "html", source = "{{ e!(//! hello) }}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^^

error: expected `)` but found `}`
 --> <source attribute>:1:16
       "}}"
  --> tests/ui/comments-in-macro-calls.rs:34:35
   |
34 | #[template(ext = "html", source = "{{ e!(//! hello)\n}}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^^^

error: missing `*/` to close block doc comment
 --> <source attribute>:1:6
       "/*! hello) }}"
  --> tests/ui/comments-in-macro-calls.rs:38:35
   |
38 | #[template(ext = "html", source = "{{ e!(/*! hello) }}")]
   |                                   ^^^^^^^^^^^^^^^^^^^^^
